modules/gwlb/main.tf

```diff
@@ -62,8 +62,8 @@ resource "aws_lb_target_group" "this" {
   tags = var.lb_target_group_tags
 
   target_failover {
-    on_deregistration = var.on_deregistration
-    on_unhealthy      = var.on_unhealthy
+    on_deregistration = var.rebalance_flows
```

Comment:
In `variables.tf` variable `rebalance_flows` is defined as `bool`.
In Terraform registry https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_target_group for [on_deregistration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_target_group#on_deregistration) and [on_unhealthy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_target_group#on_unhealthy) we have 2 possible values, which are `string`, not `bool`: 
- `rebalance`
- `no_rebalance`

Does it work for you with `bool` values?